lbrynet/p2p/HTTPBlobDownloader.py

```diff
@@ -151,6 +158,11 @@ def _download_blob(self, blob):
                 self.writers.remove(writer)
 
     def download_blob(self, blob):
+        """ Tries to download the given BlobFile object
+
+        :param BlobFile blob:
+        :return:
+        """
```

Comment:
Instead of adding the comment, can you add type annotations to the method itself

Reply:
I can do that, but again it works a lot better with reStructuredText as the docstrings so I don't see why not. It may be redundant but at least it renders well.

Reply:
Unnecessary code and comments adds to the technical debt of the project; we want to reduce technical debut. Also, we're not using sphynx doc generator. I imagine that if we do use a doc generator we'd want one that can understand Python 3 style type annotations though.

Reply:
>Unnecessary code and comments adds to the technical debt of the project; we want to reduce technical debt

I don't know what you mean by this. The entire project is written in Python, a language which sacrifices performance in exchange for readability. Not sure how an interpreter skipping comment lines is going to make it any slower.

Reply:
If you google "technical debt" you'll get a ton of articles, here are a few:

- https://blog.professorbeekums.com/all-code-is-debt/
- https://www.bmc.com/blogs/technical-debt-explained-the-complete-guide-to-understanding-and-dealing-with-technical-debt/
- https://hackernoon.com/there-are-3-main-types-of-technical-debt-heres-how-to-manage-them-4a3328a4c50c